File: python/Search/BingImageSearchv7.py
# Copyright (c) Microsoft Corporation. All rights reserved.
# Licensed under the MIT License.

# -*- coding: utf-8 -*-

import http.client
import urllib.parse
import json
import os
from pprint import pprint

'''
This sample makes a call to the Bing Image Search API with a text query and returns relevant images with data.
Documentation: https: // docs.microsoft.com/en-us/azure/cognitive-services/bing-web-search/
'''

# Add your Bing Search V7 subscription key and endpoint to your environment variables.
subscriptionKey = os.environ['BING_SEARCH_V7_SUBSCRIPTION_KEY']
host = os.environ['BING_SEARCH_V7_ENDPOINT']
host = host.replace('https://', '')
path = "/bing/v7.0/images/search"

# Query to search for
query = "puppies"

# Construct a request
headers = {'Ocp-Apim-Subscription-Key': subscriptionKey}
conn = http.client.HTTPSConnection(host)
query_search = urllib.parse.quote(query)
conn.request("GET", path + "?q=" + query_search, headers=headers)

# Print response
response = conn.getresponse()
headers = [k + ": " + v for (k, v) in response.getheaders()
                if k.startswith("BingAPIs-") or k.startswith("X-MSEdge-")]

print('Searching images for: ', query)

print("\nRelevant HTTP Headers:\n")
print("\n".join(headers))
print("\nJSON Response:\n")
pprint(json.loads(response.read()))

File: dotnet/Search/BingImageSearchv7.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using Newtonsoft.Json;
using System;
using System.Text;
using System.Net;
using System.IO;
using System.Collections.Generic;

/* This sample makes a call to the Bing Search API with a text query and returns relevant data from the web.
 * Documentation: https://docs.microsoft.com/en-us/azure/cognitive-services/bing-web-search/
 */

namespace BingImageSearch
{
    class Program
    {
        // Add your Azure Bing Search V7 subscription key and endpoint to your environment variables.
        static string subscriptionKey = Environment.GetEnvironmentVariable("BING_SEARCH_V7_SUBSCRIPTION_KEY");
        static string endpoint = Environment.GetEnvironmentVariable("BING_SEARCH_V7_ENDPOINT") + "/bing/v7.0/images/search";

        const string query = "puppies";

        static void Main()
        {
            Console.OutputEncoding = Encoding.UTF8;
            Dictionary<String, String> relevantHeaders = new Dictionary<String, String>();

            Console.WriteLine("Searching images for: " + query);

            // Construct the URI of the search request
            var uriQuery = endpoint + "?q=" + Uri.EscapeDataString(query);

            // Perform the Web request and get the response
            WebRequest request = HttpWebRequest.Create(uriQuery);
            request.Headers["Ocp-Apim-Subscription-Key"] = subscriptionKey;
            HttpWebResponse response = (HttpWebResponse)request.GetResponseAsync().Result;
            string json = new StreamReader(response.GetResponseStream()).ReadToEnd();

            // Extract Bing HTTP headers
            foreach (String header in response.Headers)
            {
                if (header.StartsWith("BingAPIs-") || header.StartsWith("X-MSEdge-"))
                    relevantHeaders[header] = response.Headers[header];
            }

            Console.WriteLine("\nRelevant HTTP Headers:\n");
            foreach (var header in relevantHeaders)
                Console.WriteLine(header.Key + ": " + header.Value);

            Console.WriteLine("\nJSON Response:\n");
            dynamic parsedJson = JsonConvert.DeserializeObject(json);
            Console.WriteLine(JsonConvert.SerializeObject(parsedJson, Formatting.Indented));
        }
    }
}
